### PR TITLE
Add example for registering callbacks with trainers

### LINKED ISSUE #9036: [docs] missing info on call back registry
https://huggingface.co/transformers/main_classes/callback.html is missing instructions/examples on how to register a callback. Thanks.

@sgugger

### PR DESCRIPTION
**Pull request by amalad**
# What does this PR do?

Fixes the issue addressed in #9036 by adding an example for registering a custom callback with the Trainer.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

Fixes: https://github.com/huggingface/transformers/issues/9036

## Who can review?

Anyone in the community is free to review the PR. But @sgugger seems the most appropriate.

### CODE REVIEW

File: docs/source/main_classes/callback.rst [hunk 1]
```diff
@@ -74,6 +74,30 @@ TrainerCallback
...
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+        callbacks=[MyCallback]
+    )
+
+Another way to register a callback is to call ``trainer.add_callback()`` as follows:
+
+.. code-block:: python
+
+    trainer.add_callback(MyCallback)
```
Comment (stas00) [inline docs/source/main_classes/callback.rst:99]: perhaps it'd help readers to have some context to when to add this? e.g.: 
```
trainer = Trainer(...)
trainer.add_callback(MyCallback)
```

**Review by stas00 — APPROVED** [review]
Looks good, thank you for adding this, @amalad

File: docs/source/main_classes/callback.rst (hunk 1 above)
Comment (amalad) [reply docs/source/main_classes/callback.rst:100]: Done. Thanks for the suggestion!

File: docs/source/main_classes/callback.rst [hunk 2]
```diff
@@ -74,6 +74,31 @@ TrainerCallback
...
+            print("Starting training")
+
+    trainer = Trainer(
+        model,
+        args,
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+        callbacks=[MyCallback]
+    )
+
```
Comment (sgugger) [inline docs/source/main_classes/callback.rst:94]: I would add a small comment here saying we can either pass the callback class (`MyCallback`) or an instance of it (`MyCallback()`).

File: docs/source/main_classes/callback.rst [hunk 3]
```diff
@@ -74,6 +74,31 @@ TrainerCallback
...
+Another way to register a callback is to call ``trainer.add_callback()`` as follows:
+
+.. code-block:: python
+
+    trainer = Trainer(...)
+    trainer.add_callback(MyCallback)
+
```
Comment (sgugger) [inline docs/source/main_classes/callback.rst:101]: And reiterate here that it can either be the class or an instance of it.

**Review by sgugger — APPROVED** [review]
Thanks you so much for doing this! I have left a few comments.

File: docs/source/main_classes/callback.rst (hunk 3 above)
Comment (amalad) [reply docs/source/main_classes/callback.rst]: Done! Does the example look better now?

**Comment by sgugger** [comment]
Thanks for updating, this looks great!

**Pull request merged**